_posts/2017-04-10-snowplow-javascript-tracker-2.8.0-released.md

```diff
@@ -0,0 +1,257 @@
+
+<h2 id="contexts">4. New OptimizelyX and Parrable contexts</h2>
+
+<h3 id="parrable">4.1 Parrable context</h3>
+
+[Parrable][parrable] is a device identification platform which
+lets you idenitfy devices across browser and apps without personally
+identifiable information. This release introduces support for a Parrable context
+which will keep track of this identifier.
+
+You can enable this context when initializing the JavaScript Tracker:
+
+{% highlight js %}
+window.snowplow('newTracker', 'cf', 'd3rkrsqld9gmqf.cloudfront.net', {
+  appId: 'CFe23a',
+  platform: 'web',
+  contexts: {
+    ...
+    'parrable': true
+  }
+});
+{% endhighlight %}
+
+You can also have a look at [the JSON Schema for the Parrable context][parrable-context].
+
+There are further setup steps required to make use of Parrable data in the Snowplow pipeline; we will provide a tutorial for this soon.
+
+Many thanks to [Parrable's Chantal Galvez Omaña][chantal] for contributing the
+Parrable support to the JavaScript Tracker!
+
+<h3 id="optimizelyx">4.2 OptimizelyX context</h3>
+
+This release also brings support for [OptimizelyX][optimizelyx], which supplements the
+support for the original Optimizely service introduced in [version 2.6.0 of the Tracker](
+/blog/2016/03/03/snowplow-javascript-tracker-2.6.0-released-with-optimizely-and-augur-integration/#optimizely-integration).
+
+You can enable this context like any other:
+
+{% highlight js %}
+window.snowplow('newTracker', 'cf', 'd3rkrsqld9gmqf.cloudfront.net', {
+  appId: 'CFe23a',
+  platform: 'web',
+  contexts: {
+    ...
+    'optimizelyXSummary': true
+  }
+});
+{% endhighlight %}
+
+The schema for this context is available [in Iglu Central][optimizelyx-context].
+
+If you are planning on leveraging the context's
+`variationName`s, please untick 'Mask descriptive names in project code
+and third-party integrations' in the OptimizelyX menu -> Settings -> Privacy.
+Otherwise, all variation names will be `null`.
+
+Thanks to [Snowflake Analytics' Mike Robins][mike] for adding the support for
+OptimizelyX.
+
+<h2 id="get-data">5. Extracting key data from the tracker</h2>
+
+In an effort to give you more control, we've exposed a few properties internal
+to the tracker such as cookie names, page view IDs and domain session indices.
+
+For example, you can retrieve the complete cookie name for the domain cookie
+with:
+
+{% highlight javascript %}
+window.snowplow_name_here('getCookieName', 'id');
+{% endhighlight %}
+
+**BEN CAN YOU EXPLAIN WHAT THE COOKIE BASENAME - HERE 'id' SIGNIFIES?**
+
+Same goes for the page view ID:
+
+{% highlight javascript %}
+window.snowplow_name_here('getPageViewId');
+{% endhighlight %}
+
+And the domain session index:
+
+{% highlight javascript %}
+window.snowplow_name_here('getDomainSessionIndex');
+{% endhighlight %}
+
+**HOW DO THESE get()s WORK IN AN ASYNC TRACKER? THEY FEEL VERY SYNCHRONOUS**
```

Comment:
Let's proceed into UA with @yalisassoon , and @chuwy and @BenFradet can explore the async gets with the tracker on the website at the same time.